cmd/internal/planetscale_edge_database.go

```diff
@@ -337,56 +374,76 @@ func (p PlanetScaleEdgeDatabase) getLatestCursorPosition(ctx context.Context, sh
 	ctx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 	var (
-		err    error
-		client psdbconnect.ConnectClient
+		err          error
+		vtgateClient vtgateservice.VitessClient
 	)
 
-	if p.clientFn == nil {
-		conn, err := grpcclient.Dial(ctx, ps.Host,
-			clientoptions.WithDefaultTLSConfig(),
-			clientoptions.WithCompression(true),
-			clientoptions.WithConnectionPool(1),
-			clientoptions.WithExtraCallOption(
-				auth.NewBasicAuth(ps.Username, ps.Password).CallOption(),
-			),
-		)
-		if err != nil {
-			return "", err
-		}
+	vtgateClient, conn, err := p.initializeVTGateClient(ctx, ps)
+	if err != nil {
+		return "", err
+	}
+	if conn != nil {
 		defer conn.Close()
-		client = psdbconnect.NewConnectClient(conn)
-	} else {
-		client, err = p.clientFn(ctx, ps)
-		if err != nil {
-			return "", err
-		}
 	}
 
-	sReq := &psdbconnect.SyncRequest{
-		TableName: s.Name,
-		Cursor: &psdbconnect.TableCursor{
-			Shard:    shard,
-			Keyspace: keyspace,
-			Position: "current",
-		},
-		TabletType: tabletType,
-		Cells:      []string{"planetscale_operator_default"},
-	}
+	vtgateReq := buildVStreamRequest(tabletType, s.Name, shard, keyspace, "current")
+	vtgateCursor, vtgateErr := vtgateClient.VStream(ctx, vtgateReq)
 
-	c, err := client.Sync(ctx, sReq)
-	if err != nil {
+	if vtgateErr != nil {
 		return "", nil
 	}
 
 	for {
-		res, err := c.Recv()
+		res, err := vtgateCursor.Recv()
 		if err != nil {
 			return "", err
 		}
 
-		if res.Cursor != nil {
-			return res.Cursor.Position, nil
+		if res.Events != nil {
+			for _, event := range res.Events {
+				if event.Type == binlogdata.VEventType_VGTID {
+					gtid := event.Vgtid.ShardGtids[0].Gtid
+					return gtid, nil
+				}
+			}
+			return "", errors.New("unable to find VEvent of VGTID type to use as stop cursor")
```

Comment:
Only find a stop position from the `VGTID` VEvent.

Reply:
@mattlord is that a safe assumption? that there will always be a `VGTID` event to get the stop position from?

Reply:
Yes, you should get the snapshot's GTID position as the second event in the first batch, for example:
```
[type:VGTID  vgtid:{shard_gtids:{keyspace:"customer"  shard:"-80"}  shard_gtids:{keyspace:"customer"  shard:"80-"  gtid:"MySQL56/17e6f390-e4a7-11ef-928d-554558eb9218:1-57"}}  keyspace:"customer"  shard:"80-"]
```

Reply:
Thank you! 🫶